src/libraries/System.Text.RegularExpressions/tests/Regex.Tests.Common.cs

```diff
@@ -24,6 +29,65 @@ public static bool IsDefaultStart(string input, RegexOptions options, int start)
             }
             return start == 0;
         }
+
+        public static IEnumerable<object[]> AvailableEngines_MemberData =>
+            from engine in AvailableEngines
+            select new object[] { engine };
+
+        public static IEnumerable<object[]> PrependEngines(IEnumerable<object[]> cases)
+        {
+            foreach (RegexEngine engine in AvailableEngines)
+            {
+                foreach (object[] additionalParameters in cases)
+                {
+                    var parameters = new object[additionalParameters.Length + 1];
+                    additionalParameters.CopyTo(parameters, 1);
+                    parameters[0] = engine;
+                    yield return parameters;
+                }
+            }
+        }
+
+        public static IEnumerable<RegexEngine> AvailableEngines
+        {
+            get
+            {
+                yield return RegexEngine.Interpreter;
+                yield return RegexEngine.Compiled;
+                if (PlatformDetection.IsNetCore &&
+                    PlatformDetection.IsReflectionEmitSupported) // the source generator doesn't use reflection emit, but it does use Roslyn for the equivalent
+                {
+                    yield return RegexEngine.SourceGenerated;
+                }
+            }
+        }
+
+        public static async Task<Regex> GetRegex(RegexEngine engine, string pattern, RegexOptions options = RegexOptions.None, int matchTimeout = -1)
+        {
+            switch (engine)
+            {
+                case RegexEngine.Interpreter:
+                    return new Regex(pattern, options, TimeSpan.FromMilliseconds(matchTimeout));
+
+                case RegexEngine.Compiled:
+                    return new Regex(pattern, options | RegexOptions.Compiled, TimeSpan.FromMilliseconds(matchTimeout));
+
+                case RegexEngine.SourceGenerated:
+                    return await RegexGeneratorHelper.SourceGenRegex(pattern, options, matchTimeout);
+            }
+
+            throw new ArgumentException($"Unknown engine: {engine}");
+        }
+
+        public static Task<Regex> GetRegex(RegexEngine engine, string pattern, RegexOptions options, TimeSpan timeout) =>
```

Comment:
nit: would it be better to use Async suffix for these methods? `GetRegexAsync`, `SourceGenRegexAsync` etc.

Reply:
Yup